BitSliceIndexing/bsi.go

```diff
@@ -39,6 +40,23 @@ func NewDefaultBSI() *BSI {
 	return NewBSI(int64(0), int64(0))
 }
 
+func (b *BSI) RunOptimize() {
+	b.eBM.RunOptimize()
+	for i := 0; i < len(b.bA); i++ {
+		b.bA[i].RunOptimize()
+	}
+	b.runOptimized = true
+}
+
+func (b *BSI) HasRunCompression() bool {
+	return b.runOptimized
+}
+
+// Returns a pointer to existence bitmap
+func (b *BSI) GetExistenceBitmap() *roaring.Bitmap {
+	return b.eBM
```

Comment:
I think in Go, comments should be of the form "GetExistenceBitmap returns a ..."

Reply:
Done